src/python/pants/bin/pants_runner.py

```diff
@@ -80,6 +83,17 @@ def run(self, start_time: float) -> ExitCode:
         ):
             # N.B. We inline imports to speed up the python thin client run, and avoids importing
             # engine types until after the runner has had a chance to set __PANTS_BIN_NAME.
+            if "SCIE" not in os.environ and "NO_SCIE_WARNING" not in os.environ:
```

Comment:
I think that @Eric-Arellano 's point was that https://github.com/pantsbuild/pants/blob/35d84ea2ab6b6d27cc1cb3ec83402b7b5f3d0e8b/src/python/pants/testutil/pants_integration_test.py#L208-L226 should be updated to disable this warning.

Reply:
So this warning won't become an error after? My understanding is that until we have a solution for installing pantsbuild.pants via pip/pex for the sake of plugin code (like linting, IDE completion, tests), we will need to release a wheel.

John in the past suggested there could be a way to use Scie to get the wheel and have Pants know how to install the dependency that way. But https://github.com/pantsbuild/pants/pull/18829 made me realize there's a simpler approach: only release the wheels for one Python version, exclusively for plugin consumption. So, we can convert this warning here to an error if you are running Pants for anything other than tests. It's annoying we'd still need to release 4 wheels, but maybe okay enough rather than figuring out this Scie idea? If we use this suggestion, then there is nothing blocking us from switching Pants to only using a single Python version, other than finishing this deprecation cycle.

Reply:
> So this warning won't become an error after? My understanding is that until we have a solution for installing pantsbuild.pants via pip/pex for the sake of plugin code (like linting, IDE completion, tests), we will need to release a wheel.

That is both correct and irrelevant here. Benjy's line of attack switches from 3.{7,8,9} wheels to 3.9 wheels only, but its still wheels as the delivery mechanism.

> It's annoying we'd still need to release 4 wheels, but maybe okay enough rather than figuring out this Scie idea?

I think you're confused here. This has nothing to do with shipping Pants as a scie. This line of attack only restricts the wheel set from 3 versions to 1 (10 wheels to 4). So I think you may be violently agreeing with this line of attack.

Shipping Pants as a scie is still blocked on work I'm doing to get the 0.1.0 of https://github.com/a-scie/lift/ released, scie-pants converted to it from using raw scie-jump, and then, finally switching from Pants wheel releases to Pants scie releases and taking the burden off PyPI.

Reply:
Perhaps the word scie-pants being the mirror image of "Pants scie" is part of the problem here. The former still uses PyPI + pip install to install Pants; so Pants must continue to be released as a wheel. The latter will not use PyPI at all and will install directly from a PEX that is released to pantsbuild/pants releases (4 PEXes - 1 per platform we support, 5 once we support Windows). Its only that latter transition that requires solving `pants_requirement` for plugin development. I'd like to solve that and not continue to use PyPI's budget as a dumping ground for a wheel we actively don't want people to discover and install (they used to "must install using ./pants" and now must install using scie-pants).

Reply:
@benjyw : If the warning is triggering in tests that are using `run_pants`, people might think that they need to pass the `NO_SCIE_WARNING` variable to `run_pants`... better to avoid that in the first place. It won't cause any test failures (yet), but it will when the deprecation triggers.